### DISCUSSION TITLE
Passing a request body in api call

### DISCUSSION THREAD

**Discussion by yogi-sagar** [artilleryio/artillery#2678] [Artillery]
Hi,
As I understand, a Payload section of the config file only supports csv file.
I am looking for a way to pass a json file content as a request body to test my api. What is the best way to achieve it?
Right now, I am sending it as below, which is working fine - 
```
scenarios:
  - name: test request api
    flow:
      - post:
            url: "/v1/TestRequests"
            beforeRequest: "setJSONBody"
            headers:
              Content-Type: application/json
              Authorization: "{{accesstoken}}"
            json: 
              "computedFormDataRequest": {
                      "Form": {
                             -----
                      }
              }       
```
However, my request body is very lengthy and complex, so I am looking for an option to pass as a json file to make the config file readable and easy to maintain.
Is it possible?

**Comment by bernardobridge** [comment]
Hi @yogi-sagar 👋 

You can use the approach described here: https://github.com/artilleryio/artillery/discussions/2638#discussioncomment-9096936
